### PR TITLE
Add linter to CI pipeline 

### PR DESCRIPTION
**Pull request by begmaroman**
- Added linter to CI pipeline
- Fixed all linter nits in the code

### CODE REVIEW

**Comment by cffls** [comment]
@begmaroman Could you hold this PR until we merge this one https://github.com/maticnetwork/heimdall/pull/842? 🙏 There will be many new changes coming from v0.3.0. It might be easier to lint everything after we merge everything from v0.3.0.

File: .github/workflows/ci.yml
Comment (JekaMas) [inline .github/workflows/ci.yml:43]: It might be a bit dangerous to run all linters against legacy code. In Bor I preferred to run linters only against changed files https://github.com/maticnetwork/bor/blob/develop/.golangci.yml#L187

Comment (begmaroman) [reply .github/workflows/ci.yml:43]: Well, you may change one file and affect another one. Could you clarify what could be dangerous?

Comment (begmaroman) [reply .github/workflows/ci.yml:43]: Oh, can I just take that config file from bor and use here?

Comment (begmaroman) [reply .github/workflows/ci.yml:43]: Done

**Review by temaniarpit27 — APPROVED** [review]
Approving because can't review 212 files

**Review by JekaMas — APPROVED** [review]
We need to test it very carefully.

**Comment by cffls** [comment]
Looks like integration tests are failing. Could you check the [logs](https://github.com/maticnetwork/heimdall/actions/runs/2577344957) and see why it failed?

**Comment by begmaroman** [comment]
@cffls fixed

**Pull request merged**
